crates/rspack_core/src/dependency/dependency_location.rs

```diff
@@ -0,0 +1,118 @@
+use std::sync::Arc;
+
+use derivative::Derivative;
+
+#[derive(Derivative)]
+#[derivative(Debug, Clone)]
+pub struct RealDependencyRange {
+  pub end: u32,
+  pub start: u32,
+  #[derivative(Debug = "ignore")]
+  source: Option<Arc<dyn SourceLocation>>,
+}
+
+impl From<swc_core::common::Span> for RealDependencyRange {
+  fn from(span: swc_core::common::Span) -> Self {
+    Self {
+      start: span.lo.0.saturating_sub(1),
+      end: span.hi.0.saturating_sub(1),
+      source: None,
+    }
+  }
+}
+
+impl RealDependencyRange {
+  pub fn new(start: u32, end: u32) -> Self {
+    RealDependencyRange {
+      end,
+      start,
+      source: None,
+    }
+  }
+
+  pub fn with_source(mut self, source: Arc<dyn SourceLocation>) -> Self {
+    self.source = Some(source);
+    self
+  }
+
+  pub fn to_loc(&self) -> Option<String> {
+    if let Some(source) = &self.source {
+      let (start, end) = source.look_up_range_pos(self.start, self.end);
+
+      if start.line == end.line {
+        if start.column == end.column {
+          return Some(format!("{}:{}", start.line, start.column));
+        }
+
+        return Some(format!("{}:{}-{}", start.line, start.column, end.column));
+      }
+
+      return Some(format!(
+        "{}:{}-{}:{}",
+        start.line, start.column, end.line, end.column
+      ));
+    }
+
+    None
+  }
+}
+
+#[derive(Debug, Clone)]
+pub struct SyntheticDependencyName {
```

Comment:
`SyntheticDependencyLocation`?

Reply:
Let's align the name if there's nothing different between us and webpack.

Reply:
It seems that this structure has been implemented in the past, now let's temporarily use this name. I will resolve the conflict issue in the next PR, which won't be too long.

Reply:
I see.